File: python/common_layer/database_models/question.py
from dataclasses import dataclass, field, asdict
from typing import List, Optional, Dict


@dataclass
class Question:
    question_id: Optional[str] = None
    topic_id: Optional[str] = None
    question: Optional[str] = None
    answers: Optional[List] = field(default_factory=list)
    explanation: Optional[str] = None
    difficulty: Optional[int] = None
    random: Optional[int] = None
    user_id: Optional[str] = None

    def prepare_for_database(self) -> dict:
        return {'question_id': self.question_id, 'topic_id': self.topic_id, 'questions': self.question,
                                 'explanation': self.explanation, 'answers': self.answers,
                                 'difficulty': self.difficulty, 'random': self.random, 'user_id': self.user_id}

File: front-end/questions-app-ui/src/app/models/question.ts
export interface QuestionHiddenCorrect {
    question_id: string;
    topic_id: string;
    questions: string;
    answers: AnswerHiddenCorrect[];
    explanation: string;
    difficulty: number;
}

export interface AnswerHiddenCorrect {
    answer: string,
    answer_id: string
}


export interface QuestionWithCorrect {
    question_id: string;
    topic_id: string;
    questions: string;
    answers: AnswerWithCorrect[];
    explanation: string;
    difficulty: number;
}

export interface AnswerWithCorrect {
    answer: string,
    answer_id: string,
    correct: boolean
}
